rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/FixtureIntersection/skip_intersection_in_union.php.inc

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddMethodCallBasedStrictParamTypeRector\FixtureIntersection;
+
+abstract class SkipIntersectionInUnion
+{
+    public function prepareParameters(): array
+    {
+        $parameters =  array_merge($this->getRequestParameters(), [
+            'request' => $this->getRequestName(),
+        ]);
+
+        $parameters = $this->removeEmptyValues($parameters);
+
+        return array_map(static fn ($parameter) => 'test', $parameters);
+    }
+
+    private function removeEmptyValues($input): array
+    {
+        foreach ($input as &$value) {
+            if (! is_array($value)) {
+                continue;
+            }
+
+            $value = $this->removeEmptyValues($value);
+        }
+
+        return array_filter($input, static fn ($val) => true);
+    }
+}
```

Comment:
This looks pretty complex. Is there a way to make this fixture simpler to only contain the intersection type?

Reply:
updated with reduce complexity, that's the only code that reproduce the error https://github.com/rectorphp/rector-src/pull/2704/commits/dc54a426e029e9020e7a452dc976261718427ea4